build-deploy/common/models/build.js

```diff
@@ -63,30 +66,60 @@ module.exports = function(Build) {
       break;
     }
 
+    debug('spawn: %j', args);
+
     var buildProcess = spawn(process.execPath, args);
+
     processes[build.id] = buildProcess;
 
     build.handleOutput('out', buildProcess.stdout);
     build.handleOutput('err', buildProcess.stderr);
 
-    buildProcess.on('exit', function(errCode) {
+    buildProcess.on('error', function(err) {
+      debug('spawn errored: %s', err);
+
+      // Not a code... but best we can do without changing Build schema.
+      var errorCode = err.message;
+
+      finish(build.id, errorCode);
+    });
+
+    buildProcess.on('exit', function(status, signal) {
+      var errorCode = signal || status;
+
+      debug('child %d exit: %s', buildProcess.pid, errorCode);
+
+      // Don't set on status 0, that's success, anything else is an error.
+      if (!errorCode) {
+        errorCode = undefined;
+      }
+
+      finish(build.id, errorCode, build._parseArchivePath());
+    });
+
+    function finish(id, errorCode, archive) {
       delete processes[build.id];
-      Build.findById(build.id, function(err, build) {
-        if(err) {
```

Comment:
@anthonyettinger I replaced this attempt to deal with `err` with an assert, because if there is an `err`, then build would be `undefined`, and attempting to write `build.error` would throw an exception and die, anyhow.

But its pretty crappy, is there any better way to deal with errors here, or is this good enough? It probably never happens, except when arc backend is killed unexpectedly.